src/Microsoft.Diagnostics.Monitoring.RestServer/ActionContextExtensions.cs

```diff
@@ -14,47 +16,79 @@ namespace Microsoft.Diagnostics.Monitoring.RestServer
 {
     internal static class ActionContextExtensions
     {
+        private const string ExceptionLogMessage = "Request failed.";
+
         public static Task ProblemAsync(this ActionContext context, Exception ex)
         {
-            ActionResult result = new BadRequestObjectResult(ex.ToProblemDetails((int)HttpStatusCode.BadRequest));
+            if (context.HttpContext.Features.Get<IHttpResponseFeature>().HasStarted)
```

Comment:
How come this is necessary? Aren't we always await'ing response writes?

Reply:
Because the exception was thrown while processing the action of the OutputStreamResult or EgressStreamResult, the action could have written data to the response stream already. If we try to rewrite response stream with whatever the exception information is, it will throw the InvalidOperationException.

I believe this frequently happens for anything that is streaming information back through the response stream and it is cancelled; for example, the logs endpoint.